src/test/java/com/google/api/codegen/packagegen/testdata/library.proto

```diff
@@ -64,40 +87,61 @@ service LibraryService {
   // This call is a no-op if the specified shelves are the same.
   rpc MergeShelves(MergeShelvesRequest) returns (Shelf) {
     option (google.api.http) = { post: "/v1/{name=shelves/*}:merge" body: "*" };
+    option (google.api.method_signature) = {
+      fields: ["name", "other_shelf_name"]
+    };
   }
 
   // Creates a book, and returns the new Book.
   rpc CreateBook(CreateBookRequest) returns (Book) {
     option (google.api.http) = { post: "/v1/{name=shelves/*}/books" body: "book" };
+    option (google.api.method_signature) = {
+      fields: ["name", "book"]
+    };
   }
 
   // Gets a book. Returns NOT_FOUND if the book does not exist.
   rpc GetBook(GetBookRequest) returns (Book) {
     option (google.api.http) = { get: "/v1/{name=shelves/*/books/*}" };
+    option (google.api.method_signature) = {
+      fields: ["name"]
+    };
   }
 
   // Lists books in a shelf. The order is unspecified but deterministic. Newly
   // created books will not necessarily be added to the end of this list.
   // Returns NOT_FOUND if the shelf does not exist.
   rpc ListBooks(ListBooksRequest) returns (ListBooksResponse) {
     option (google.api.http) = { get: "/v1/{name=shelves/*}/books" };
+    option (google.api.method_signature) = {
+      fields: ["name"]
+    };
   }
 
   // Deletes a book. Returns NOT_FOUND if the book does not exist.
   rpc DeleteBook(DeleteBookRequest) returns (google.protobuf.Empty) {
     option (google.api.http) = { delete: "/v1/{name=shelves/*/books/*}" };
+    option (google.api.method_signature) = {
+      fields: ["name"]
+    };
   }
 
   // Updates a book. Returns INVALID_ARGUMENT if the name of the book
   // is non-empty and does equal the previous name.
   rpc UpdateBook(UpdateBookRequest) returns (Book) {
     option (google.api.http) = { put: "/v1/{name=shelves/*/books/*}" body: "book" };
+    option (google.api.method_signature) = {
+      fields: ["name", "book", "update_mask"]
```

Comment:
This seems wrong. I would have expected either `["book"]` or `["book", "update_mask"]` (probably the latter).